[bad merge] py-pyprecice: Add version 2.1.1.2 and 2.2.0.1

ajaust opened the pull request:
This adds the updated Python bindings for the current versions preCICE (https://github.com/spack/spack/pull/19264 and https://github.com/spack/spack/pull/21166)

The installation procedure has been improved such that we need the patch `deactivate-version-check-via-pip.patch` only for versions `<= 2.1.1.1` of the bindings. This is reflected accordingly in the Spack recipe.

ajaust committed:
add preCICE 2.1.1.2 and 2.2.0.1 to package

ajaust committed:
add preCICE 2.1.1.2 and 2.2.0.1 to package

bvanessen committed:
lbann: updated python dependencies and new variants (#21863)

* Made DiHydrogen a required dependencies on newer versions of LBANN.
Added an explicit variant for enabling Boost-dependent callbacks.
Updated the separation for embedded Python and the Python front end
code and associated dependencies.

* Bugfix on ROCm include in DiHydrogen

scheibelp committed:
reduce strictness of directory layout spec-equality check (#21869)

scheibelp committed:
"spack build-env" searches env for relevant spec (#21642)

If you install packages using spack install in an environment with
complex spec constraints, and the install fails, you may want to
test out the build using spack build-env; one issue (particularly
if you use concretize: together) is that it may be hard to pass
the appropriate spec that matches what the environment is
attempting to install.

This updates the build-env command to default to pulling a matching
spec from the environment rather than concretizing what the user
provides on the command line independently.

This makes a similar change to spack cd.

If the user-provided spec matches multiple specs in the environment,
then these commands will now report an error and display all
matching specs (to help the user specify).

Co-authored-by: Gregory Becker <becker33@llnl.gov>

ketsubouchi committed:
redland bindings: new package at v1.0.17.1 (#21838)

nhanford committed:
New splice method in class Spec. (#20262)

* Spec.splice feature

Construct a new spec with a dependency swapped out. Currently can only swap dependencies of the same name, and can only apply to concrete specs.

This feature is not yet attached to any install functionality, but will eventually allow us to "rewire" a package to depend on a different set of dependencies.

Docstring is reformatted for git below

Splices dependency "other" into this ("target") Spec, and return the result as a concrete Spec.

If transitive, then other and its dependencies will be extrapolated to a list of Specs and spliced in accordingly.

For example, let there exist a dependency graph as follows:

        T
        | \
        Z<-H

In this example, Spec T depends on H and Z, and H also depends on Z.

Suppose, however, that we wish to use a differently-built H, known as H'. This function will splice in the new H' in one of two ways:

1. transitively, where H' depends on the Z' it was built with, and the new T* also directly depends on this new Z', or
2. intransitively, where the new T* and H' both depend on the original Z.

Since the Spec returned by this splicing function is no longer deployed the same way it was built, any such changes are tracked by setting the build_spec to point to the corresponding dependency from the original Spec.

Co-authored-by: Nathan Hanford <hanford1@llnl.gov>

ptooley committed:
use package supplied autogen.sh (#20319)

joshessman-llnl committed:
Updates to support clingo-cffi (#20657)

* Support clingo when used with cffi

Clingo recently merged in a new Python module option based on cffi.

Compatibility with this module requires a few changes to spack - it does not automatically convert strings/ints/etc to Symbol and clingo.Symbol.string throws on failure.

manually convert str/int to clingo.Symbol types
catch stringify exceptions
add job for clingo-cffi to Spack CI
switch to potassco-vendored wheel for clingo-cffi CI
on_unsat argument when cffi

zzzoom committed:
julia: Add 1.5.3 (#21891)

RemiLacroix-IDRIS committed:
Sph2pipe: Update URL (#21886)

dorton21 committed:
New version r-stringfish at 0.14.2 (#21785)

scottwittenburg committed:
Gitlab fix pr workflow (#21786)

Fixes for gitlab pipelines

* Remove accidentally retained testing branch name
* Generate pipeline w/out debug mode
* Make jobs interruptible for auto-cancel pending
* Work around concretization conflicts

dorton21 committed:
New package r-fastdigest at 0.6-3 (#21778)

dorton21 committed:
New package r-circstat@0.2-6 (#21777)

rblake-llnl committed:
numactl: no-op needed if configure can't find TLS (#21894)

* numactl: Bash needs no-op if configure can't find TLS

* Upstream has fix, will be out in next version.

https://github.com/numactl/numactl/pull/103

haampie committed:
HIP: document variables, apply patches to older versions, fix build issues (#21852)

Since there are many variables being set I thought it would be a good idea to document them better and slightly simplify the logic for external vs not-external.

haampie committed:
Fix target selection in rocfft (#21858)

kuramoto-fj committed:
n2p2: Add new package (#21709)

* n2p2: Add new package

* remove ,

* Resurrection of , and changed " to single

* changed example.command to example.co

* n2p2: Added v2.1.1

* n2p2: Changed the type of depends_on.

adamjstewart commented on var/spack/repos/builtin/packages/py-pyprecice/package.py:
```diff
@@ -44,7 +51,7 @@ class PyPyprecice(PythonPackage):
     depends_on("py-mpi4py", type=("build", "run"), when="+mpi")
```
I won't think the `wheel` dep is needed

adamjstewart replied:
Also, the `py-mpi4py` package is always required, so I would remove the variant.

adamjstewart commented on var/spack/repos/builtin/packages/py-pyprecice/package.py:
```diff
@@ -44,7 +51,7 @@ class PyPyprecice(PythonPackage):
     depends_on("py-cython@0.29:", type=("build"))
 
-    phases = ['build_ext', 'install']
+    phases = ['install_lib', 'build_ext', 'install']
```
Why is this change needed?

BenjaminRodenberg replied:
We switched from a single python module (only [`precice` in v2.1.1.2 and older](https://github.com/precice/python-bindings/blob/b5a2c82d4577cd637f7fdf63f68e3a4ccb1746a2/setup.py#L137)) to having two modules:

* [`precice`](https://github.com/precice/python-bindings/blob/d5383d6e4be2424aacbb0d24ab1e9c2a8212713b/setup.py#L151) is the actual user interface
* [`cyprecice`](https://github.com/precice/python-bindings/blob/d5383d6e4be2424aacbb0d24ab1e9c2a8212713b/setup.py#L79-L86) provides the actual implementation

This was necessary, since we started using `versioneer` to maintain versions (https://github.com/precice/python-bindings/pull/70) and this could not be done directly in the cython extension.

I tried building the spack module without the `install_lib` phase, but this was not successful. Then I [added this stage](https://github.com/precice/python-bindings/pull/68/commits/cc893289bb97c426c5c48a74fdb2fa6fda5cbdff) and it worked.

adamjstewart commented on var/spack/repos/builtin/packages/py-pyprecice/package.py:
```diff
@@ -53,4 +60,5 @@ def build_ext_args(self, spec, prefix):
         ]
 
     def install(self, spec, prefix):
-        self.setup_py("install", "--prefix={0}".format(prefix))
+        if self.version <= Version("2.1.1.1"):
+            self.setup_py("install", "--prefix={0}".format(prefix))
```
Does the default install method not work for some reason?

BenjaminRodenberg replied:
We had to use the non-default method due to https://github.com/precice/python-bindings/issues/64 in older versions. In Version `2.2.0.1` this issue is fixed, but we still have to use the non-default method for compatibility reasons. Or is there a better way how to do this in spack?

Here is the discussion from our original PR: https://github.com/spack/spack/pull/19558#discussion_r513123239.

BenjaminRodenberg commented on var/spack/repos/builtin/packages/py-pyprecice/package.py:
```diff
@@ -53,4 +60,5 @@ def build_ext_args(self, spec, prefix):
 
     def install(self, spec, prefix):
-        self.setup_py("install", "--prefix={0}".format(prefix))
+        if self.version <= Version("2.1.1.1"):
```
```suggestion
        if self.version <= Version("2.1.1.1"):  # Needed for compatibility reasons. See https://github.com/spack/spack/pull/21881
```

BenjaminRodenberg replied:
Maybe a comment here will help us in the future?

Paul-Ferrell committed:
Config prefer upstream (#21487)

This allows for quickly configuring a spack install/env to use upstream packages by default. This is particularly important when upstreaming from a set of officially supported spack installs on a production cluster. By configuring such that package preferences match the upstream, you ensure maximal reuse of existing package installations.

dorton21 committed:
New package r-rapiserialize at 0.1.0 (#21780)

eugeneswalker committed:
rempi-needs-libpciaccess (link dep) (#21652)

eugeneswalker committed:
libcircle: needs libpciaccess (link dep) (#21648)

adamjstewart committed:
py-azureml-dataprep-rslex: add new package (#21915)

adamjstewart committed:
py-azureml-automl-core: add new version (#21916)

adamjstewart committed:
py-azureml-core: add new version (#21917)

adamjstewart committed:
py-azureml-dataprep-native: add new version (#21918)

adamjstewart committed:
py-azureml-dataset-runtime: add new version (#21920)

adamjstewart committed:
py-azureml-pipeline-core: add new version (#21921)

adamjstewart committed:
py-azureml-pipeline-steps: add new version (#21922)

adamjstewart committed:
py-azureml-pipeline: add new version (#21923)

adamjstewart committed:
py-azureml-sdk: add new version (#21924)

adamjstewart committed:
py-azureml-telemetry: add new version (#21925)

adamjstewart committed:
py-azureml-train-automl-client: add new version (#21926)

adamjstewart committed:
py-azureml-train-core: add new version (#21927)

adamjstewart committed:
py-azureml-train-restclients-hyperdrive: add new version (#21928)

adamjstewart committed:
py-azureml-train: add new version (#21929)

alalazo committed:
Run clingo-cffi tests in a container (#21913)

There clingo-cffi job has two issues to be solved:

1. It uses the default concretizer
2. It requires a package from https://test.pypi.org/simple/

The former can be fixed by setting the SPACK_TEST_SOLVER
environment variable to "clingo".

The latter though requires clingo-cffi to be pushed to a
more stable package index (since https://test.pypi.org/simple/
is meant as a scratch version of PyPI that can be wiped at
any time).

For the time being run the tests in a container. Switch back to
PyPI whenever a new official version of clingo will be released.

RemiLacroix-IDRIS committed:
Silo: Update URLs (#21880)

cessenat committed:
Silo: fix build errors with +silex on ubuntu18.04(#21877)

dorton21 committed:
New package r-qs at 0.23.5 (#21779)

adamjstewart committed:
py-azureml-dataprep: add new version (#21919)

adamjstewart committed:
font-util: fix typo in default fonts (#21933)

mvanwaveren committed:
libjpeg: add version 9d (#21819)

* Update package libjpeg

* Change order and remove empty line.

* emove space

adamjstewart committed:
Overhaul xgboost packages (#21661)

* Overhaul xgboost packages
* Don't deprecate xgboost 0.90
* Add py-xgboost@0.90, but deprecate it
* extras_require added in 1.0.0
* Deprecate py-dask-xgboost
* Simplify setting cuda arch

eugeneswalker committed:
llvm-doe+lldb: @doe: fix improper set of LLDB_ENABLE_PYTHON (#21935)

dorton21 committed:
New package r-spades-core at 1.0.5 (#21782)

giordano committed:
libblastrampoline: Add version 3.0.0 (#21939)

adamjstewart committed:
py-webob: new package at v1.8.7 (#21949)

adamjstewart committed:
py-h5netcdf: new package at v0.10.0 (#21947)

adamjstewart committed:
py-cfgrib: new package at v0.9.8.5 (#21946)

adamjstewart committed:
py-numcodecs: add v0.7.3, apple clang patch (#21944)

* py-numcodecs: add new version, apple clang patch

* py-numcodecs: add new version, apple clang patch

haampie committed:
kcov: add missing dependency (#21932)

kcov CMakeLists.txt generates the "kcov" executable only if
certain dependencies are found. These dependencies are 
"libbfd", "libopcodes" and "libiberty", hence the dependency 
on binutils.

haampie committed:
libxc: add perl as a build dependency (#21906)

haampie committed:
openblas: add perl as a build dependency (#21908)

haampie committed:
Update fuse packages (#21901)

Allow libfuse to build without setuid binary and bump versions of both
libfuse and fuse-overlayfs.

Still doesn't solve the issue where this package tries to install things
into /etc/init.d though.

ronin-gw committed:
w3m: new package at v0.5.3 (#21641)

smithsg84 committed:
gridlab-d: specify xerces prefix (#21867)

iarspider committed:
Fix recipe for evtgen 2.0; add variant for hepmc3 (#21814)

h-denpo committed:
libpulsar: added dependency on curl (#21950)

cessenat committed:
harfbuzz: dependency assumed cairo was compiled with +pdf and +ft (#21954)

adamjstewart committed:
py-pydap: add new package (#21948)

ax3l committed:
WarpX: CMake (#21787)

* New Package: py-picmistandard

* New Package: py-warpx

* WarpX: Transition to CMake

haampie committed:
openmpi: add perl as a build dependency (#21956)

michaelkuhn committed:
meson: add 0.57.1 (#21959)

skosukhin committed:
hdf5: drop support for libtool files (#21900)

eugeneswalker committed:
tioga: add variants: +cuda and more (#21884)

aweits committed:
py-tensorflow: disable h5py/hdf5 mpi variant(s) if tensorflow is ~mpi (#21955)

dorton21 committed:
New package r-spades-addins at 0.1.2 (#21781)

vchuravy committed:
[flux-core] add -Wno-errpr=maybe-uninitialized (#21943)

Co-authored-by: Stephen Herbein <SteVwonder@users.noreply.github.com>

mkuron committed:
Old concretizer: prevent unexpected propagation of external config (#20976)

When using an external package with the old concretizer, all
dependencies of that external package were severed. This was not
performed bidirectionally though, so for an external package W with
a dependency on Z, if some other package Y depended on Z, Z could
still pull properties (e.g. compiler) from W since it was not
severed as a parent dependency.

This performs the severing bidirectionally, and adds tests to
confirm expected behavior when using config from DAG-adjacent
packages during concretization.

DarylGrunau committed:
pegtl: manage options to build examples and tests when run_tests (#21866)

Co-authored-by: Daryl W. Grunau <dwg@lanl.gov>

becker33 committed:
typo fix (#21967)

glennpj committed:
Revert "New package r-circstat@0.2-6 (#21777)" (#21974)

This reverts commit 61c8bc2b19941bb009bf261337765a2475a42459.

lee218llnl committed:
added waylaynd-protocols package (#21979)

healther committed:
make `spack fetch` work with environments (#19166)

* make `spack fetch` work with environments
* previously: `spack fetch` required the explicit statement of
              the specs to be fetched, even when in an environment
* now: if there is no spec(s) provided to `spack fetch` we check
       if an environment is active and if yes we fetch all
       uninstalled specs.

ketsubouchi committed:
cpmd: add expected smoke test check output (#21668)

haampie committed:
CMake: add a "build_type" variant (#21958)

This permits to use MinSizeRel or other build types
where they are a better fit.

eugeneswalker committed:
hypre: enable cuda variants (#21885)

h-denpo committed:
dbow2: add a dependency on eigen (#21663)

eschnett committed:
hwloc: add v2.4.1 (#21978)

ax3l committed:
openPMD-api: use cmake helpers (#21973)

Modernize the `openpmd-api` package to use fancy new `CMakePackage`
helpers when setting options from variants :-)

adamjstewart committed:
py-pooch: add new package (#21972)

adamjstewart committed:
py-importlib-resources: add v5.1.0 (#21970)

adamjstewart committed:
Python: add new versions (#21969)

adamjstewart committed:
c-blosc: add v1.21.0 (#21968)

adamjstewart committed:
py-xarray: add new variant 'io' (#21945)

haampie committed:
berkeley-db: do not install docs by default (#21960)

* Add an option to not install 100MB of docs

* Add deprecations of versions that are removed and add conflicts for 5.3

t-nojiri committed:
warpx: fix typos in a few dependency conditions (#21982)

eugeneswalker committed:
amr-wind: new package (#21883)

* amr-wind: new package

* amr-wind: simplify cuda_arch mapping

* simplify

t-nojiri committed:
ncl: allow building on aarch64 (#21984)

danlipsa committed:
sensei: add versions and fix dependencies (#21452)

* Fix sensei@develop

Should work with all options but libsim.
Current releases don't work with ~catalyst
See
https://gitlab.kitware.com/sensei/sensei/-/merge_requests/240
for the fix for develop.
Current releases work only with paraview 5.7 and 5.6
See
https://gitlab.kitware.com/sensei/sensei/-/merge_requests/239
for the fix for develop (which works with 5.9)

* Fix libsim.

* Fix warnings.

* Fix python runtime.

* Many changes:

* Reworked cmake options top use the CMakePackage option helpers
* Simplified and consolidated options
* Replaced adios with adios2 variant
* Added vtkm variant (not yet working)

* paraview: Fix downstream consumers getting the wrong FindMPI

* vtk: Fix downstream consumers getting the wrong FindMPI

* Add +ascent, +adios2; remove +adios; variants off by default

* Fix catalyst python logic

* sensei: cleanup formatting

Co-authored-by: Chuck Atkins <chuck.atkins@kitware.com>

adamjstewart committed:
xkbcomp: add v 1.4.4 (#21981)

cessenat committed:
asciidoc: sourceforge a2x needs python2, adds more recent for py3 (#21807)

* asciidoc: current sourceforge a2x needs python2, new github release python3

* asciidoc: current sourceforge a2x needs python2, new github release python3

* asciidoc: making python 2.3 to 2.7 able to cope with asciidoc

* asciidoc: current sourceforge a2x needs python2, new github release python3

* asciidoc: current sourceforge a2x needs python2, new github release python3

* asciidoc: current sourceforge a2x needs python2, new github release python3

* asciidoc: current sourceforge a2x needs python2, new github release python3

cessenat committed:
Qscintilla: added v2.12.0, fixed build (#21902)

adamjstewart committed:
py-metpy: add new package (#21971)

iarspider committed:
FastJet: fix patch level (#21936)

dorton21 committed:
Package/r circstats (#21966)

mic84 committed:
amrex: +hypre and +petsc do not need +fortran from version 20.08 (#21994)

eugeneswalker committed:
amrex: use right cmake_args for @develop; depends_on(hypre) for link (#21892)

* amrex package: use right cmake_args for @develop; depends_on(hypre)

* add explicit depends_on for petsc, for link, when +petsc

lee218llnl committed:
Compilers: make oneapi ifx version_regex tolerant to wrappers (#21742)

If a user creates a wrapper for the ifx binary called ifx_orig,
this causes the ifx --version command to produce:

$ ifx --version
ifx_orig (IFORT) 2021.1 Beta 20201113
Copyright (C) 1985-2020 Intel Corporation. All rights reserved.

The regex for ifx currently expects the output to begin with
"ifx (IFORT)..." so the wrapper would not be detected as ifx. This
PR removes the need for the static "ifx" string which allows wrappers
to be detected as ifx.

In general, the Intel compiler regexes do not include the invoked
executable name (i.e., ifort, icc, icx, etc.), so this is not
expected to cause any issues.

dorton21 committed:
New package r-spades-tools at 0.3.6 (#21783)

rviertel committed:
adds flag to turn off NLOPT_OCTAVE in CMAKE when -octave (#21997)

aweits committed:
nccl-tests: respect cuda_arch variant during build (#21995)

stmobo committed:
boost: Patch B2 bootstrap step to use correct toolchain (#21408)

balay committed:
petsc/petsc4py/slepc: switch default git branch from 'master' to 'main' (#22001)

adamjstewart committed:
kcov: elfutils not needed on macOS (#21993)

tldahlgren committed:
Raise InstallError if any explicit specs is skipped (#21980)

iarspider committed:
Openblas: add experimental BIGNUMA flag (#21393)

iarspider committed:
Rivet updates and improvements (#21843)

ajaust committed:
update precice bindings

Remove MPI dependency
Making mpi4py dependency explicit
Making install phases version dependend such that we only need two phases for installation
Adding some comments to explain dependencies and patches

ajaust committed:
update precice bindings

Remove MPI dependency
Making mpi4py dependency explicit
Making install phases version dependend such that we only need two phases for installation
Adding some comments to explain dependencies and patches

ajaust committed:
Merge branch 'update-py-pyprecice' of github.com:ajaust/spack into update-py-pyprecice

adamjstewart commented:
Looks like a bad merge. Might want to make a new PR.

ajaust commented:
Yeah, seems like I did something wrong. Sorry for the noise. We will fix the recipe and reopen the PR.
